File: products/shop_branding/src/shoot.js
const { chromium } = require('playwright');
const path = require('path');

const SRC = __dirname;
const OUT = path.join(__dirname, '..');

const jobs = [
  { file: 'logo.html', out: 'logo.png', w: 500, h: 500 },
  { file: 'banner-big.html', out: 'banner-big.png', w: 3360, h: 840 },
  { file: 'banner-mini.html', out: 'banner-mini.png', w: 1600, h: 213 },
];

(async () => {
  const browser = await chromium.launch();
  for (const j of jobs) {
    const page = await browser.newPage({
      viewport: { width: j.w, height: j.h },
      deviceScaleFactor: 1,
    });
    await page.goto('file://' + path.join(SRC, j.file));
    await page.waitForTimeout(150);
    await page.screenshot({ path: path.join(OUT, j.out) });
    await page.close();
    console.log('wrote', j.out);
  }
  await browser.close();
})();

File: products/shop_branding/src/shoot.py
import os
from pathlib import Path
from playwright.sync_api import sync_playwright

SRC = Path(__file__).parent
OUT = SRC.parent

jobs = [
    ("logo.html", "logo.png", 500, 500),
    ("banner-big.html", "banner-big.png", 3360, 840),
    ("banner-mini.html", "banner-mini.png", 1600, 213),
]

with sync_playwright() as p:
    browser = p.chromium.launch(executable_path="/opt/pw-browsers/chromium")
    for file, out, w, h in jobs:
        page = browser.new_page(viewport={"width": w, "height": h}, device_scale_factor=1)
        page.goto((SRC / file).as_uri())
        page.wait_for_timeout(150)
        page.screenshot(path=str(OUT / out))
        page.close()
        print("wrote", out)
    browser.close()
